### PR TITLE
utils/temporary: the --temp-dir does not work

### PR DESCRIPTION
**Pull request by kshtsk**
Using of --temp-dir argument does not make an effect,
because optional 'dir' parameter defaults to the
global TEMP_DIR value before it's changed.

This patch address this issue.

Signed-off-by: Kyr Shatskyy <kyrylo.shatskyy@gmail.com>

### CODE REVIEW

**Comment by schaefi** [comment]
You are so right, shame on me, this is a bad one. Thanks much for fixing

File: kiwi/utils/temporary.py
Comment (schaefi) [inline kiwi/utils/temporary.py:34]: As we are on it, using the built-in name "dir" was not a good idea from me as well. Codacy is also complaining about it. Could we rename the parameter e.g ```dir_name``` or ```tmp_dir_name```

Comment (schaefi) [reply kiwi/utils/temporary.py:34]: I realized changing the parameter name will impact some calls, but I think this would be a good change

Comment (kshtsk) [reply kiwi/utils/temporary.py:34]: It is a `path` so maybe using it instead of `dir_name`?
Like in description `Provides namespace to handle temporary files and directories`.

Comment (schaefi) [reply kiwi/utils/temporary.py:34]: Fine by me

**Comment by schaefi** [comment]
I've approved the workflow run, once you change the parameter from "dir" to "path" there will be some errors.

Just in case you don't know you can run the tests and static type checks locally with:

```
tox -e check,unit_py3_[YourMajorPythonVersion] "-n 5"
```

**Comment by schaefi** [comment]
Thanks much

**Comment by kshtsk** [comment]
> I've approved the workflow run, once you change the parameter from "dir" to "path" there will be some errors.
> 
> Just in case you don't know you can run the tests and static type checks locally with:
> 
> ```
> tox -e check,unit_py3_[YourMajorPythonVersion] "-n 5"
> ```
```
tox -e 'check,unit_py3_[YourMajorPythonVersion]' "-n 5"
ERROR: unknown environment 'unit_py3_[YourMajorPythonVersion]'
```

**Comment by kshtsk** [comment]
ah...stupid me, I should replace with version number, hm... I have only 3.9 at my system :-D no worries, I will sort it out

File: kiwi/utils/temporary.py
Comment (kshtsk) [reply kiwi/utils/temporary.py:34]: @schaefi I took another look at the file, it looks like a wrapper around `tempfile.NamedTemporaryFile` and `tempfile.TemporaryDirectory`, which is using `dir` arguments in constructor, so it makes sense to leave the older signature too.

Comment (schaefi) [reply kiwi/utils/temporary.py]: yes the methods from the tempfile module uses ```dir``` arguments, which we just have to accept as it is

**Review by schaefi — APPROVED** [review]
Looks great to me :+1: Let's wait for the tests to pass and we are ready for merging

**Comment by schaefi** [comment]
> ah...stupid me, I should replace with version number, hm... I have only 3.9 at my system :-D no worries, I will sort it out

Adding 3.9 and maybe even 3.10 targets to tox.ini sounds like a good idea too. I'll look into that once we have merged this PR

**Pull request merged**
